notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1200 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "d5d3e2e38df7"
+   },
+   "outputs": [],
+   "source": [
+    "mono_ranking_avg_bytes = (\n",
+    "    preprocessed_df.groupBy(\"mono_language\")\n",
+    "    .agg(\n",
+    "        count(\"mono_language\").alias(\"count\"),\n",
+    "        round(avg(\"mono_size\") / 1000).alias(\"average_in_MB\"),\n",
+    "    )\n",
+    "    .sort(desc(\"average_in_MB\"))\n",
+    "    .where(col(\"mono_language\").isNotNull() & (col(\"count\") > 500))\n",
+    ")\n",
+    "\n",
+    "mono_ranking_avg_bytes.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "4551e1b7b28b"
+   },
+   "source": [
+    "#### What three languages are most frequently found together in polyglot repos?\n",
+    "\n",
+    "Using the preprocessed `poly_language` column, implement a query to show the ranking of polyglot repositories with the top 3 languages, based on size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "c8285a340a4a"
+   },
+   "outputs": [],
+   "source": [
+    "# Get the polyglot repositories by the language popularity.\n",
+    "poly_ranking = (\n",
+    "    preprocessed_df.groupBy(\"poly_language\")\n",
+    "    .count()\n",
+    "    .sort(desc(\"count\"))\n",
+    "    .where(col(\"poly_language\").isNotNull())\n",
+    ")\n",
+    "\n",
+    "poly_ranking.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "633a0752b053"
+   },
+   "source": [
+    "The majority of results contain a combination of HTML or CSS, and Javascript.\n",
+    "\n",
+    "Display a pie chart?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "30ae2d750ab7"
+   },
+   "outputs": [],
+   "source": [
+    "# Convert the Spark DataFrame to a Pandas DataFrame to display the pie chart.\n",
+    "poly_panda = poly_ranking.toPandas()[:LIMIT].copy()\n",
+    "poly_panda.groupby([\"poly_language\"]).sum().plot(\n",
+    "    kind=\"pie\",\n",
+    "    y=\"count\",\n",
+    "    autopct=\"%1.1f%%\",\n",
+    "    label=\"\",\n",
+    "    title=\"Polyglot repositories\",\n",
+    "    legend=False,\n",
+    "    figsize=(7, 7),\n",
+    "    explode=EXPLODE_PIE_CHART,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "3905037bcea3"
+   },
+   "source": [
+    "The pie chart shows that eight out of the top ten results contain either `HTML` or `CSS`.\n",
+    "\n",
+    "You can create combinations of languages in each repo using the original data fetched from BigQuery."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "25479dcebd93"
+   },
+   "outputs": [],
+   "source": [
+    "# A Python package to get combinations.\n",
+    "from itertools import combinations\n",
```

Comment:
if itertools is a package/function name, put in `code` format. Otherwise please explain or set context.